### ISSUE TITLE
There's no head element, still it complains that it's missing the title element.

### ISSUE THREAD

**Issue by tbytedev** [validator/validator#1668]
HTML code
```
<!DOCTYPE html>
<html lang="en">
Some text
</html>
```
Error message:
```
Element head is missing a required instance of child element title.
```
There's no head element, so it can't miss anything.

**Comment by Madhusri02** [comment]
Actually this code does't show any error when i workout . As it still shows the error for you, I can conclude that it requires <head> tag as it includes the metadata . check out the code with <head> tag

**Comment by tbytedev** [comment]
I don't say I fully understand what you wrote, but I think this error occurs not only for me, but for everyone who visits https://validator.w3.org/. I'm sorry, but I forgot to mention, that I typed that short html code there. And there was a link to this github repo, so that if I notice any errors, I can report them here. So I assumed that it's not necessary to mention. And I don't understand what metadata you are referring.

**Comment by scottaohara** [comment]
the title element is required in most cases.

while the head element isn't required - that's because browsers will automatically inject it for you.

so you can do the following:

```
<!DOCTYPE html>
<html lang="en">
<title>Test</title>
<body>
....
</body>
</html>
```

to both not have to have a head element, but still have the required title element that the browser will make sure is placed into the head element they inject.

**Comment by tbytedev** [comment]
The reason I made this issue is that I think the error message is incorrect. So the problem was not that I couldn't make a valid HTML file, and I need some help how to correct my 4 line long code. I made that code deliberately to demonstrate how to reproduce that error message. As the code doesn't contain head element the validator shouldn't say that the head element is missing something. It should just say that the title tag is required.

**Comment by scottaohara** [comment]
appreciate the clarification @tbytedev, as it was not clear that you didn't understand this from your original comments -
specifically:
>There's no head element, so it can't miss anything.

that reads to me that you're saying because a head isn't present, the title isn't required.  Not, "this error message is confusing because even though the title is required, it's implying i need to declare a head element, which i don't"

and to that point i generally agree with you.  It would be good to get the error message updated so it could more generally state that regardless of whether a head element is present, a title element still need to be specified by the author.

whether that be done by updating the output to state that the document's implicitly added head element is missing the title element.  Or, to maybe instead use the more general `REQUIRED_ELEMENTS_MISSING = "Required elements missing.".toCharArray();` rather than the missing required child message, to mitigate the mention of the implicit or explicit head element entirely.

**Comment by sideshowbarker** [comment]
> ```
> <!DOCTYPE html>
> <html lang="en">
> Some text
> </html>
> ```
> 
> ```
> Element head is missing a required instance of child element title.
> ```
> 
> There's no head element, so it can't miss anything.

Open that source in a browser devtools and what you’ll see is that there is in fact a `head` element in that document. (And what’s more, there’s also a `body` element in that document.)

It’s important to understand this.

It’s true that in the _source_ above there is no `<head>` start tag nor `</head>` end tag — but the `head` _element_ is still there.

It’s important to understand the difference between elements and tags. For some elements, the start tags and end tags can be omitted. For the `head` element, both the start tag and the end tag can be omitted.

But even in a document where the `head` element’s start tag or end tag are both omitted, that document still has a `head` element. _Every_ HTML document has a `head` element — even if the `head` element has no contents (like the one above).

Consider the following document served as `text/html`:
```html
Some text
```
That’s the entire source for the document. It has no markup in that source — no tags at all. But if it’s served as `text/html` and parsed by an HTML parser, it in fact contains three elements: an `html` element, a `head` element, and `body` element.

If you want confirmation of that fact:

1. Create a completely empty file named `foo.html` (or whatever)
2. Put just the text `Some text` into that file
3. Open that `foo.html` file in a browser (any browser).
4. Open the inspect in devtools in the browser.
5. Look at what it shows.

If you do the steps above, you’ll see that the HTML document represented by the source above is identical to the document represented by this source:
```html
<html><head></head><body>Some text
</body></html>
```
The start tags and end tags for the `html`, `head`, and `body` elements are all optional. So the only difference between the source in the above two cases is that in the first case, the start tags and end tags have been completely omitted. But it’s still the same document.

In other words, for understanding HTML documents, it’s important — and even necessary — to not just consider the source but to instead consider what an HTML parser does with that source when it creates an actual HTML document from it.

It’s also important to understand that the HTML checker (validator) doesn’t just look at the source it’s given; instead the checker first parses that source (just as browsers do) and then checks the resulting HTML document the parser creates.

So, what the HTML checker “sees” (and what browsers see) doesn’t look like the tags-omitted source in the first case above but instead looks like the source in the second case above, with all the optional tags included.

In yet other words, when looking at some HTML markup/source, it’s necessary to understand what HTML document that source _represents_ — which minimally includes understanding that _every_ HTML document has `html`, `head`, and `body` elements, even if the source for the document doesn’t have start tags or end tags for any of those elements.
